docs/administration.building-blocks.md

```diff
@@ -0,0 +1,127 @@
+  but manual building blocks can be a good start to get familiar with the concept and give a clear ordering flow
+  to your customers. It could also be an option to automate the building block in the future.
+- Terraform: this is the option to automate your building block. You will only need a Terraform file and meshStack will
+  run this configuration everytime a user books a new building block. In a later step you can setup Inputs for your
+  Terraform script. There is also a [section](#developing--deploying-a-terraform-building-block)
+  below with further tips for developing Terraform building blocks.
+
+If you pick Terraform you have to enter a few further values:
+
+- Terraform Version: the version your Terraform files are based on.
+- Git Repository URL: the URL of the Git repository where your Terraform files reside.
+- (Optional) Git Repository Path: the specific directory in the repository where the Terraform files reside.
+- (Optional) Private SSH Key: upload an SSH key that has access to the provided Git repository.
+
+Lastly, you have to decide what the behavior of the building block is when it is deleted by the customer user. This can be
+
+- Purge in meshStack: (this is the only option if you have a manual building block) The building block will be deleted
+  in meshStack, but no underlying resources (e.g. in a cloud platform) will be deleted. You have to do this yourself.
+- Delete Resources: this option will also clean up the underlying resources. The workings depend on the implementation
+  type, but e.g. for Terraform meshStack will call `terraform destroy` on the configuration files.
+
+### Dependencies
+
+> Please note that this the use of dependencies is completely optional and this step can be skipped.
+
+In some cases you might find that you have a multiple building block definitions, and one depends on the other.
+For example, you might have a Virtual Network, which is a prerequisite for a Virtual Machine.
+
+You can model and enforce such dependencies in meshStack using the Building Blocks feature.
+
+In the wizard, on the "Dependencies" page, you have to select the building block(s) that the _about to be created_
+building block depends on. So in the example above you would have to do the following:
+
+1. Create the Virtual Network Building Block without any dependencies.
+2. Create the Virtual Machine Building Block and select the Virtual Network as a dependency.
+
+> For Terraform users: you can provide the output of the Virtual Network Building Block as input for the Virtual
+> Machine building block.
+
+### Building Block Inputs
+
+As with a lot of configuration, input is required. Building Blocks in meshStack can collect as much input as you want,
+in many ways possible.
+
+There are different kinds of inputs, with different behavior, which can be decided under the "Source" section. You can
+read in the meshPanel what each kind of input does and how it behaves.
```

Comment:
d: I think it would also be worth mentioning those typoes here for readers of the docs who are just inetrested in meshStack, but don't use it yet. With all those input types we can create some wow moments on what is possible here